velox/functions/lib/Re2Functions.cpp

```diff
@@ -738,6 +745,7 @@ std::shared_ptr<exec::VectorFunction> makeLike(
       name,
       inputArgs[1].type->toString());
 
+  bool hasEscape = false;
```

Comment:
nit: setting `hasEscape = numArgs == 3` and using `if (hasEscape)` might be clearer